src/bin/taskmasterd.rs

```diff
@@ -1,12 +1,44 @@
-use std::io::{Read, Write};
+use daemonize::Daemonize;
 use std::{
     convert::TryFrom,
+    fs::File,
+    io::{Read, Write},
     net::{TcpListener, TcpStream},
-    path::Path,
+    path::{Path, PathBuf},
+    process,
 };
 use taskmaster::{command::Command, config::Config, DEFAULT_ADDR};
+use users::{get_current_gid, get_current_uid};
+
+fn daemonize(home: &PathBuf) {
+    let stderr = File::create(home.join("taskmasterd.log")).unwrap();
+
+    let daemonize = Daemonize::new()
+        .pid_file(home.join("taskmasterd.pid"))
+        .chown_pid_file(true)
+        .working_directory(home)
+        .user(get_current_uid())
+        .group(get_current_gid())
+        .umask(0o027)
+        .stderr(stderr)
+        .privileged_action(|| "Executed before drop privileges");
+
+    match daemonize.start() {
+        Ok(_) => (),
+        Err(e) => {
+            eprintln!("Error, {}", e);
+            process::exit(1)
+        }
+    }
+}
 
 fn main() -> Result<(), std::io::Error> {
+    let dir = dirs::home_dir();
```

Comment:
To avoid having to rationalize the use of an external crate, do you think we could use `std::env::var_os("HOME")` ? :)

This is basically what `dirs` [does first](https://github.com/dirs-dev/dirs-sys-rs/blob/e776c08a220bdb38cc64b29d7912fdee42143343/src/lib.rs#L31-L35).
I mean sure it won't work if HOME is not set, but maybe we can just error of of that ?

Reply:
Sure, no prob ! I'll fix that too.